deployer/tasks/setup.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Deployer;
+
+use Deployer\Exception\RuntimeException;
+use Deployer\Task\Context;
+use Symfony\Component\Yaml\Yaml;
+
+desc('Setup redaxo instance');
+task('setup', new class {
+    public function __invoke()
+    {
+        $this->checkConfigYml();
+    }
+
+    private function checkConfigYml()
+    {
+        cd('{{release_path}}');
+
+        if (test('[ -f {{data_dir}}/core/config.yml ]')) {
+            return;
+        }
+
+        writeln('');
+        writeln('<comment>Setting config.yml</comment>');
+        writeln('');
+
+        $config = Yaml::parse(file_get_contents(getcwd().'/'.get('data_dir').'/core/config.yml'));
+
+        $config['setup'] = false;
+        $config['debug'] = false;
+        $config['instname'] = 'rex'.date('YmdHis');
+
+        $config['server'] = ask('Server:', 'https://'.Context::get()->getHost()->getRealHostname());
+        $config['servername'] = ask('Server name:', $config['servername']);
+        $config['error_email'] = ask('Error email:', $config['error_email']);
+
+        writeln('');
+
+        $db = $config['db'][1];
+        $db['host'] = 'localhost';
+        $db['login'] = 'root';
+        $db['password'] = null;
+        $db['name'] = null;
+
+        $dbValid = false;
+
+        do {
+            $db['host'] = ask('Database host:', $db['host']);
+            $db['login'] = ask('Database user:', $db['login']);
+            $db['password'] = askHiddenResponse('Database password:');
+            $db['name'] = ask('Database name:', $db['name']);
+
+            try {
+                run("mysql -h$db[host] -u$db[login] -p".escapeshellarg($db['password'])." $db[name] -e ';'");
+                $dbValid = true;
+            } catch (RuntimeException $e) {
+                writeln('');
+                writeln('<error>Could not connect to database: '.$e->getErrorOutput().'</error>');
+                writeln('');
+            }
+        } while (!$dbValid);
+
+        $config['db'][1] = $db;
+
+        $config = Yaml::dump($config, 3);
+
+        run('mkdir -p {{data_dir}}/core');
+        run('echo '.escapeshellarg($config).' > {{data_dir}}/core/config.yml');
```

Comment:
Das ist vermutlich auch noch nicht gut gelöst, oder?
Auch hier steckt ja das Passwort direkt im Command dann..

Also vielleicht besser lokal die Config in einer Temp-Datei ablegen und dann per `upload()` hochladen?

Reply:
für mich macht das keinen unterschied... dass passwort ist in beiden wegen im klartext auf deiner workstation vorhanden.

vllt interessant in diesem zusammenhang https://github.com/deployphp/deployer/issues/1527

Reply:
Mir ging es eher um den Server, dass dort das Passwort eventuell in `ps` oder irgendwo in der history oder so dann auftaucht.
Aber das ist Quatsch bzw. egal?

Reply:
Jop, diesbzgl dann besser mit file (und kommentar im source)